Bottom: 707dd6e442696cd42c51f57adf989f482ed735b5
Top:    8089253aa2b78288d6d9df0da1fa17661083c343
Author: Mark de Wever <koraq@xs4all.nl>
Date:   2021-08-07 14:31:57 +0200

Refresh of add-a-calculator-controller

---

diff --git a/tests/calculator/controller.cpp b/tests/calculator/controller.cpp
index 7a9f418..5b82357 100644
--- a/tests/calculator/controller.cpp
+++ b/tests/calculator/controller.cpp
@@ -23,14 +23,7 @@ namespace calculator {
 TEST(controller, constructor) {
   tmodel model;
   const tcontroller controller{model};
-  EXPECT_TRUE(noexcept(tcontroller{model}));
-
-  //
-  //
-  //
-  //
-  //
-  // XXX Backport to other tests..
+  static_assert(noexcept(tcontroller{model}));
 }
 
 TEST(controller, copy_constructor) {
@@ -45,7 +38,7 @@ TEST(controller, move_constructor) {
 
 TEST(controller, destructor) {
   tmodel model;
-  EXPECT_TRUE(noexcept(tcontroller{model}.~tcontroller()));
+  static_assert(noexcept(tcontroller{model}.~tcontroller()));
 }
 
 TEST(controller, copy_assignment) {
@@ -61,7 +54,7 @@ TEST(controller, move_assignment) {
 TEST(controller, push) {
   tmodel model;
   tcontroller controller{model};
-  EXPECT_TRUE(noexcept(controller.push()));
+  static_assert(noexcept(controller.push()));
 }
 
 static std::string format_error(const char *message) {
